pkg/clustermodule/service.go

```diff
@@ -34,12 +35,28 @@ func NewService() Service {
 }
 
 func (s service) Create(ctx *context.ClusterContext, wrapper Wrapper) (string, error) {
-	logger := ctx.Logger.WithValues("object", wrapper.GetName())
-	template, err := fetchMachineTemplate(ctx, wrapper)
+	logger := ctx.Logger.WithValues("object", wrapper.GetName(), "namespace", wrapper.GetNamespace())
+
+	templateRef, err := fetchTemplateRef(ctx, ctx.Client, wrapper)
+	if err != nil {
+		logger.V(4).Error(err, "error fetching template for object")
+		return "", errors.Wrapf(err, "error fetching machine template for object %s/%s", wrapper.GetNamespace(), wrapper.GetName())
+	}
+	if templateRef.Kind != "VSphereMachineTemplate" {
```

Comment:
Better to use a constant value instead of hard coded string "VSphereMachineTemplate"